src/main/java/com/hltech/pact/gen/domain/pact/PactFactory.java

```diff
@@ -209,8 +211,21 @@ private static InteractionResponse createInteractionResponse(
     }
 
     private static Map<String, String> mapHeaders(List<Param> headers) {
+        Map<String, List<String>> mappedHeadersWithDuplicates = mapHeadersWithDuplicates(headers);
+
+        if (mappedHeadersWithDuplicates.values().stream().anyMatch(list -> list.size() > 1)) {
+            log.error("Only single value for given header allowed. Single value will be assigned");
```

Comment:
What about add more precise info about which values are duplicated?

Reply:
Also - shouldn't it be better to throw exception in such case?

Reply:
It is against JSON rule to duplicate keys. Pacts are expressed as JSONs. I was not sure whether to throw exception here or not as maybe we can treat as "any from set". Maybe I can add entry in README.